main.go

```diff
@@ -74,58 +74,77 @@ func extractFromTag(tag *ast.BasicLit) (string, string) {
 }
 
 // exprToSchema converts ast.Expr to JSONSchemaProps
-func exprToSchema(t ast.Expr, comment string, importPaths map[string]string, curPkgPrefix string) (*v1beta1.JSONSchemaProps, []TypeReference) {
-	def := &v1beta1.JSONSchemaProps{}
+func (f *file) exprToSchema(t ast.Expr, doc string, comments []*ast.CommentGroup) (*v1beta1.JSONSchemaProps, []TypeReference) {
+	var def *v1beta1.JSONSchemaProps
 	var externalTypeRefs []TypeReference
-	var typeName, pkgAlias string
 
 	switch tt := t.(type) {
 	case *ast.Ident:
-		typeName = tt.Name
+		def = f.identToSchema(tt, doc, comments)
 	case *ast.ArrayType:
-		def, externalTypeRefs = arrayTypeToSchema(tt, comment, importPaths, curPkgPrefix)
+		def, externalTypeRefs = f.arrayTypeToSchema(tt, doc, comments)
 	case *ast.MapType:
-		def = mapTypeToSchema(tt, curPkgPrefix)
+		def = f.mapTypeToSchema(tt, doc, comments)
 	case *ast.SelectorExpr:
-		selectorType := tt
-		pkgAlias = selectorType.X.(*ast.Ident).Name
-		typeName = selectorType.Sel.Name
-		def.Ref = getPrefixedDefLink(typeName, importPaths[pkgAlias])
-		externalTypeRefs = []TypeReference{{TypeName: typeName, PackageName: pkgAlias}}
+		def, externalTypeRefs = f.selectorExprToSchema(tt, doc, comments)
 	case *ast.StarExpr:
-		def, externalTypeRefs = exprToSchema(tt.X, comment, importPaths, curPkgPrefix)
+		def, externalTypeRefs = f.exprToSchema(tt.X, "", comments)
 	case *ast.StructType:
-		def, externalTypeRefs = structTypeToSchema(tt, importPaths, curPkgPrefix)
+		def, externalTypeRefs = f.structTypeToSchema(tt, comments)
+	case *ast.InterfaceType: // ignore it, since we don't care about it.
```

Comment:
Can we always ignore this? Will there be a scenario when this needs to be parsed?

Reply:
If so, adding a TODO would be better.

Reply:
Done. Added a TODO.